dmutils/filters.py

```diff
@@ -24,18 +25,19 @@ def format_links(text):
                                 \.                        # match dot in host name
                                 (?:[^\s/?\.#<>"']+\.?)+   # match rest of domain
                                 (?:/[^\s<>"']*)?          # rest of url
-                                [^\s<>"'\.]               # no dot at end
+                                [^\s<>,"'\.]              # no dot at end
                                 )""", re.X)
-    matches = url_match.findall(text)
-    link = '{0}<a href={1} rel="external">{1}</a>'
-    if matches:
-        textArray = url_match.split(text)
+    matched_urls = url_match.findall(text)
+    if matched_urls:
+        link = '{0}<a href="{1}" rel="external">{2}</a>'
         formattedText = ""
-        for text in textArray:
-            if text in matches:
-                formattedText = link.format(formattedText, Markup.escape(text))
+        textArray = url_match.split(text)
+        for partial_text in textArray:
+            if partial_text in matched_urls:
+                url = "http://{}".format(partial_text) if partial_text.startswith('www') else partial_text
```

Comment:
This bit is interesting since it means we're linking to HTTP by default even if there's an HTTPS version, which is not ideal. I don't think we can do the reverse either, since it'll likely break some links, while HTTP link has a chance of being promoted to HTTPS.

Can we avoid linking `www.` at all and just look for `http[s]://` so that we don't end up with broken links? I don't know how common `www.` is nowadays, and we're not looking for any other URL/domain formats.